# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mplayer/x11-monitor-aspect.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Set the monitor aspect automatically for X displays as done by other drivers,
e.g. the OS X one. Formerly 4:3 was always assumted leading to obviously
false apsect on recent 16:9 / 16:10 desktop and laptop displays.

  - Rene Rebe <rene@exactcode.de>

--- MPlayer-1.0pre7try2/libvo/x11_common.c	2005-04-16 17:52:03.000000000 +0200
+++ MPlayer-1.0pre7try2-fixed/libvo/x11_common.c	2006-02-12 19:04:15.070424750 +0100
@@ -87,6 +87,8 @@
 static int vo_fs_flip = 0;
 char **vo_fstype_list;
 
+extern float monitor_aspect;
+
 /* if equal to 1 means that WM is a metacity (broken as hell) */
 int metacity_hack = 0;
 
@@ -510,6 +512,14 @@
            vo_screenwidth, vo_screenheight, depth, vo_depthonscreen,
            dispName, mLocalDisplay ? "local" : "remote");
 
+    {
+        int w, h;
+        w = XDisplayWidthMM (mDisplay, mScreen);
+        h = XDisplayHeightMM (mDisplay, mScreen);
+        if (w && h)
+            monitor_aspect = (float) w / h;
+    }
+
     vo_wm_type = vo_wm_detect();
 
     vo_fs_type = vo_x11_get_fs_type(vo_wm_type);
